pkg/http-handler/http-handler.go

```diff
@@ -111,22 +110,12 @@ func UpdateAppRepository(handler kube.AuthHandler) func(w http.ResponseWriter, r
 func ValidateAppRepository(handler kube.AuthHandler) func(w http.ResponseWriter, req *http.Request) {
 	return func(w http.ResponseWriter, req *http.Request) {
 		token := auth.ExtractToken(req.Header.Get("Authorization"))
-		res, err := handler.AsUser(token).ValidateAppRepository(req.Body, mux.Vars(req)["namespace"])
+		err := handler.AsUser(token).ValidateAppRepository(req.Body, mux.Vars(req)["namespace"])
 		if err != nil {
-			returnK8sError(err, w)
-			return
-		}
-		body, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			returnK8sError(err, w)
+			w.Write([]byte(err.Error()))
```

Comment:
Wouldn't this mean that we're now parsing the response body and string matching to determine an error?